CHANGELOG.md

```diff
@@ -9,14 +9,48 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 ## [Unreleased]
 
 ### Added
+- Adds a custom MarshalJSON method for `trace.Link` to marshal all fields. (#1852)
 
-- - Extract resource attributes from the `OTEL_RESOURCE_ATTRIBUTES` environment variable and merge them with the `resource.Default` resource as well as resources provided to the `TracerProvider` and metric `Controller`. (#1785)
```

Comment:
This CHANGELOG is not properly merged. It needs to be sync-ed with main and only contain the changes relevant to this PR.